### PR TITLE
fix: addresses .env not being loadable be new devenv

### LINKED ISSUE #52: [Bug]:  The dotenv file '.env' was not found after updating to edge.24-06-2
### Prior Search

- [X] I have already searched this [project's issues](https://github.com/Panfactum/stack/issues) to determine if a bug report has already been made.

### What happened?

At some point between `2048c3439e17a497a947495129639c8b38d4a6b9` (edge.2024-05-23) and `58b63910959cf791b313f836236ecdbefd2bc644` (edge.2024-06-02), the nix flake was no longer able to find the correct path for `$DEVENT_ROOT/.env` 

This renders the `.env` file nonoperative and any functionality depending on it, e.g. `VAULT_ADDR`, `VAULT_TOKEN`, `AWS_PROFILE` is broken because these values are now `@INVALID@` or unset.

### Steps to Reproduce

`nix flake update`

### Version

main (development branch)

### Relevant log output

```shell
❯ direnv reload
direnv: loading ~/Projects/corgeek.net/stack/.envrc
direnv: loading https://raw.githubusercontent.com/nix-community/nix-direnv/3.0.4/direnvrc (sha256-DzlYZ33mWF/Gs8DDeyjr8mnVmQGx7ASYqA5WlxwvBG4=)
direnv: using flake . --impure
warning: Git tree '/home/matt/Projects/corgeek.net/stack' is dirty
evaluating derivation 'git+file:///home/matt/Projects/corgeek.net/stack#devShells.x86_64-linux.default'direnv: ([direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
warning: Git tree '/home/matt/Projects/corgeek.net/stack' is dirty
direnv: nix-direnv: renewed cache
💡 The dotenv file '.env' was not found.
```

### PR DESCRIPTION
**Pull request by mschnee**
Fixes #52 

According to devenv maintainers, this is a __wontfix__, 

While the old python-rewrite branch supported dotenv files when used via nix flakes, the released 1.0.x rewrite in rust does not. https://github.com/cachix/devenv/issues/1135#issuecomment-2127681361

TLDR, using the released devenv means `.env` files no longer work.

### CODE REVIEW

**Comment by fullykubed** [comment]
Thank you for this! Used it as inspiration for the fix, but added the changes in a different commit b/c needed to make some other changes as well.
